internal/venice-common/src/test/java/com/linkedin/venice/servicediscovery/TestServiceDiscoveryAnnouncerRetryTask.java

```diff
@@ -0,0 +1,102 @@
+    doReturn(30L * Time.MS_PER_SECOND).when(config).getServiceDiscoveryRegistrationRetryMS();
+    long serviceDiscoveryRegistrationRetryMS = config.getServiceDiscoveryRegistrationRetryMS();
+    List<ServiceDiscoveryAnnouncer> serviceDiscoveryAnnouncers = Arrays.asList(announcer1, announcer2, announcer3);
+    AsyncRetryingServiceDiscoveryAnnouncer asyncRetryingServiceDiscoveryAnnouncer =
+        new AsyncRetryingServiceDiscoveryAnnouncer(serviceDiscoveryAnnouncers, serviceDiscoveryRegistrationRetryMS);
+    BlockingQueue<ServiceDiscoveryAnnouncer> retryQueue =
+        asyncRetryingServiceDiscoveryAnnouncer.getServiceDiscoveryAnnouncerRetryQueue();
+    try {
+      asyncRetryingServiceDiscoveryAnnouncer.register();
+    } catch (Exception e) {
+      Assert.fail("The method call should not throw an exception.");
+    }
+
+    Assert.assertTrue(retryQueue.contains(announcer1));
+    Assert.assertTrue(retryQueue.contains(announcer2));
+    Assert.assertFalse(retryQueue.contains(announcer3));
+    Assert.assertEquals(retryQueue.peek(), announcer1);
+    Assert.assertEquals(retryQueue.size(), 2);
+
+    putTestToSleep(serviceDiscoveryRegistrationRetryMS + 1000);
+    Assert.assertTrue(retryQueue.contains(announcer1));
+    Assert.assertTrue(retryQueue.contains(announcer2));
+    Assert.assertEquals(retryQueue.peek(), announcer2);
+    Assert.assertEquals(retryQueue.size(), 2);
+
+    putTestToSleep(serviceDiscoveryRegistrationRetryMS + 1000);
+    Assert.assertTrue(retryQueue.contains(announcer1));
+    Assert.assertFalse(retryQueue.contains(announcer2));
+    Assert.assertEquals(retryQueue.peek(), announcer1);
+    Assert.assertEquals(retryQueue.size(), 1);
+
+    putTestToSleep(serviceDiscoveryRegistrationRetryMS + 1000);
+    Assert.assertFalse(retryQueue.contains(announcer1));
+    Assert.assertEquals(retryQueue.size(), 0);
+  }
+
+  @Test
+  public void testUnregisterServiceDiscoveryAnnouncers() {
+    reset(announcer1, announcer2, announcer3, config);
+    doThrow(new RuntimeException()).when(announcer1).unregister();
+    doNothing().when(announcer2).unregister();
+    doNothing().when(announcer3).unregister();
+    doReturn(30L * Time.MS_PER_SECOND).when(config).getServiceDiscoveryRegistrationRetryMS();
+    long serviceDiscoveryRegistrationRetryMS = config.getServiceDiscoveryRegistrationRetryMS();
+    List<ServiceDiscoveryAnnouncer> serviceDiscoveryAnnouncers = Arrays.asList(announcer1, announcer2, announcer3);
+    AsyncRetryingServiceDiscoveryAnnouncer asyncRetryingServiceDiscoveryAnnouncer =
+        new AsyncRetryingServiceDiscoveryAnnouncer(serviceDiscoveryAnnouncers, serviceDiscoveryRegistrationRetryMS);
+    try {
+      asyncRetryingServiceDiscoveryAnnouncer.unregister();
+    } catch (Exception e) {
+      Assert.fail("The method call should not throw an exception.");
+    }
+  }
+
+  private void putTestToSleep(long ms) {
+    LOGGER.info("Test sleeping for {} ms", ms);
+    sleep(ms);
+    LOGGER.info("Test woke up");
+  }
```

Comment:
We don't need this additional function and logging. Let's just replace this with `Utils.sleep` directly